sdk/core/core/src/az_http_request.c

```diff
@@ -148,7 +148,7 @@ az_http_request_append_header(_az_http_request* p_request, az_span key, az_span
 {
   AZ_PRECONDITION_NOT_NULL(p_request);
   AZ_PRECONDITION_VALID_SPAN(key, 1, false);
-  AZ_PRECONDITION_VALID_SPAN(value, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(value, 0, true);
```

Comment:
For my edification, can you please provide some context on why this should be allowed? What is motivating this change?

Reply:
headers with no value are valid on http requests.
See the issue description. Current precondition block us from adding  a header with no value.
See the unit tests I added as well

Reply:
> headers with no value are valid on http requests.

Is there a spec or doc that I can read about that?

I found this but it doesn't mention the value can be optional.
https://developer.mozilla.org/en-US/docs/Glossary/Request_header

Reply:
this is the RFC spec: https://tools.ietf.org/html/rfc7230#section-3.2